tests/Operation/ReplaceOneTest.php

```diff
@@ -48,4 +50,25 @@ public function testConstructorReplacementArgumentProhibitsUpdatePipeline($repla
         $this->expectExceptionMessageMatches('#(\$replacement is an update pipeline)|(Expected \$replacement to have type "document" \(array or object\))#');
         new ReplaceOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], $replacement);
     }
+
+    /** @dataProvider provideInvalidConstructorOptions */
+    public function testConstructorOptionsTypeCheck($options): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        new ReplaceOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], ['y' => 1], $options);
+    }
+
+    public function provideInvalidConstructorOptions()
+    {
+        return $this->createOptionDataProvider([
+            'codec' => $this->getInvalidDocumentCodecValues(),
+        ]);
+    }
+
+    public function testCodecRejectsInvalidDocuments(): void
+    {
+        $this->expectExceptionObject(UnsupportedValueException::invalidEncodableValue([]));
+
+        new ReplaceOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], ['y' => 1], ['codec' => new TestDocumentCodec()]);
```

Comment:
Noted this fails because TestDocumentCodec only supports TestObject and an array is being passed.